### PR TITLE
fix: Support get segments from current segments view

### PR DESCRIPTION
**Pull request by xiaocai2333**
issue: #38511

### CODE REVIEW

**Comment by xiaocai2333** [comment]
rerun go-sdk

File: internal/datacoord/handler.go [hunk 1]
```diff
@@ -263,6 +272,124 @@ func (h *ServerHandler) GetQueryVChanPositions(channel RWChannel, partitionIDs .
 	}
 }
 
+func (h *ServerHandler) GetCurrentSegmentsView(ctx context.Context, channel RWChannel, partitionIDs ...UniqueID) *SegmentsView {
```
Comment (bigsheeper) [inline internal/datacoord/handler.go:275]: Use `GetQueryVChanPositions` to prevent duplicated codes.

Comment (xiaocai2333) [reply internal/datacoord/handler.go:283]: It's not exactly the same as `GetQueryVChanPositions`, I have extracted the common functions.

File: internal/datacoord/services.go [hunk 2]
```diff
@@ -1044,6 +1044,15 @@ func (s *Server) GetSegmentsByStates(ctx context.Context, req *datapb.GetSegment
 	} else {
 		segmentIDs = s.meta.GetSegmentsIDOfPartition(ctx, collectionID, partitionID)
 	}
```
Comment (liliu-z) [inline internal/datacoord/services.go:1046]: Dup？

**Pull request merged**